### PR TITLE
support for amazon ec2

### PR DESCRIPTION
**Pull request by will-chan**

### CODE REVIEW

File: handlers/create.go [hunk 1]
```diff
@@ -209,6 +209,44 @@ func buildMachineCreateCmd(machine *client.Machine) ([]string, error) {
...
+			cmd = append(cmd, "--amazonec2-security-group", machine.Amazonec2Config.SecurityGroup)
+		}
+		if machine.Amazonec2Config.InstanceType != "" {
+			cmd = append(cmd, "--amazonec2-instance-type", machine.Amazonec2Config.InstanceType)
+		}
+		if machine.Amazonec2Config.RootSize != "" {
+			cmd = append(cmd, "--amazonec2-root-size", machine.Amazonec2Config.RootSize)
+		}
+		if machine.Amazonec2Config.IamInstanceProfile != "" {
+			cmd = append(cmd, "--amazonec2-access-key", machine.Amazonec2Config.AccessKey)
```
Comment (cjellick) [inline handlers/create.go]: Whoops! Should not be access key.

**Comment by cjellick** [comment]
@will-chan This change will _require_ 0.2.0rc of docker-machine to work because of new ec2 parameters, correct?

That feels like a bit of a gamble since you mentioned 0.2.0rc also comes with a change to the directory structure which would impact other functionality like how we extract the config.

Feel confident you can pull off the upgrade and ec2 functionality in time for the demo?

**Comment by cjellick** [comment]
FWIW, based on whats left in the 0.2.0 docker-machine milestone, I do feel comfortable that we can run with the RC. It's all documentation and build/ci related stuff.

https://github.com/docker/machine/milestones/0.2.0

File: handlers/create.go (hunk 1 above)
Comment (will-chan) [inline handlers/create.go]: Good catch.  Will fix it.

**Comment by will-chan** [comment]
It may be supported in 0.2 but there were issues with running it.  I can submit as a separate PR in case we need to revert.  The minimum fields work right now.

**Comment by cjellick** [comment]
Ok, I'm going to restate because I want to make sure I understand what you're suggesting and that we are on the same page.

The field `--amazonec2-iam-instance-profile` is not in machine v0.1.0, but we do have code for it.

You're proposing that we keep the field in the rancher API, but don't expose it in the UI until we upgrade to v0.2.0 (which we will do separately from the ec2 code), right?

The risk/impact is that if `iamInstanceProfile` _is_ sent down from rancher and we are still on machine 0.1.0, the machine will fail to deploy.

The odds of that are extremely low, so I'm ok with doing it, but we could completely eliminate the risk by simply commenting out the code that adds `--amazonec2-iam-instance-profile` to the exec command while leaving it in the rancher API. That way we wont have to mess with schemas or go-rancher when we bump to 0.2.0, we'll just have to uncomment the translation code.

**Comment by will-chan** [comment]
@cjellick  That's correct. I will create a separate PR for 0.2 support so we can back out of it if it's just broken.  It has directory changes.  For now, iam-instance-profile just won't be supported.

Also, I just added better error parsing from docker logs and send that back up to cattle rather than the generic exit status code which is what docker-machine will return usually when it fails.  Review that for me.

**Comment by will-chan** [comment]
I also read the code and docker-machine will just ignore any flags it doesn't know about anyways.  I tried it and nothing happens so even if the UI is there, it won't really do any harm right now. The machine will still get created as long as the other fields are correctly set.

**Comment by cjellick** [comment]
Oh, I tried it in 0.1.0 and it blew up for me if I passed an unknown arg.

At any rate, I'm ok with leaving it as is.

**Comment by will-chan** [comment]
Let's leave it.  I'll quickly add 0.2 support this upcoming week anyways.

File: handlers/create.go [hunk 2]
```diff
@@ -98,7 +103,7 @@ func CreateMachine(event *events.Event, apiClient *client.RancherClient) error {
 	return publishReply(reply, apiClient)
 }
 
-func logProgress(readerStdout io.Reader, readerStderr io.Reader, publishChan chan<- string, uuid string, event *events.Event) {
+func logProgress(readerStdout io.Reader, readerStderr io.Reader, publishChan chan<- string, uuid string, event *events.Event, errString *string) {
```
Comment (cjellick) [inline handlers/create.go]: Do you need a string pointer here or could just a normal string work?

**Comment by cjellick** [comment]
For future reference, a PR doesnt _have_ to be a single commit. So, the error parsing improvements should have been a separate commit. (Really, it should have been its own PR, but no big deal in this case.)

**Pull request merged**